### PR TITLE
refactor(autoware_compare_map_segmentation): resolve clang-tidy error in autoware_compare_map_segmentation

### PR DESCRIPTION
**Pull request by yhisaki**
## Description

Resolved errors warned by clang-tidy checks.

## Related links

- [TIER IV Slack Link](https://star4.slack.com/archives/C070FQMREVB/p1730089933677489?thread_ts=1729747263.248229&cid=C070FQMREVB)

## Notes for reviewers

None.

## Interface changes

None.

## Effects on system behavior

None.

### CODE REVIEW

File: perception/autoware_compare_map_segmentation/src/voxel_grid_map_loader/voxel_grid_map_loader.hpp [hunk 1]
```diff
@@ -93,13 +92,15 @@ class VoxelGridMapLoader
   bool debug_ = false;
 
 public:
-  typedef VoxelGridEx<pcl::PointXYZ> VoxelGridPointXYZ;
-  typedef typename pcl::Filter<pcl::PointXYZ>::PointCloud PointCloud;
-  typedef typename PointCloud::Ptr PointCloudPtr;
+  using VoxelGridPointXYZ = VoxelGridEx<pcl::PointXYZ>;
+  using PointCloud = typename pcl::Filter<pcl::PointXYZ>::PointCloud;
```
Comment (knzo25) [inline perception/autoware_compare_map_segmentation/src/voxel_grid_map_loader/voxel_grid_map_loader.hpp:96]: @badai-nguyen 
(This is unrelated to this PR but appeared as a change so I will leave a message)
This seems to be a typo.

Comment (yhisaki) [reply perception/autoware_compare_map_segmentation/src/voxel_grid_map_loader/voxel_grid_map_loader.hpp:96]: Thank you, I will fix it.
Where should I change it?

File: sensing/autoware_pointcloud_preprocessor/include/autoware/pointcloud_preprocessor/filter.hpp [hunk 2]
```diff
@@ -132,6 +132,8 @@ class Filter : public rclcpp::Node
     const std::string & filter_name = "pointcloud_preprocessor_filter",
     const rclcpp::NodeOptions & options = rclcpp::NodeOptions());
 
+  ~Filter() override;
```
Comment (knzo25) [inline sensing/autoware_pointcloud_preprocessor/include/autoware/pointcloud_preprocessor/filter.hpp:135]: Sorry, maybe I am just not understanding enough,  but should not be a definition for this declaration ? or default or similar?

Comment (yhisaki) [reply sensing/autoware_pointcloud_preprocessor/include/autoware/pointcloud_preprocessor/filter.hpp:135]: Sorry, this change may not have solved the underlying problem.
The problem is described in following message.
```
/opt/ros/humble/include/message_filters/message_filters/subscriber.h:111:3: error: all paths through this function will call itself [clang-diagnostic-infinite-recursion]
  {
  ^
/opt/ros/humble/include/message_filters/message_filters/subscriber.h:205:3: note: in instantiation of member function 'message_filters::SubscriberBase<>::subscribe' requested here
  ~Subscriber()
  ^
/home/hisaki/workspace/tmp/autoware/install/autoware_pointcloud_preprocessor/include/autoware_pointcloud_preprocessor/autoware/pointcloud_preprocessor/filter.hpp:104:7: note: in instantiation of member function 'message_filters::Subscriber<sensor_msgs::msg::PointCloud2_<std::allocator<void>>>::~Subscriber' requested here
class Filter : public rclcpp::Node
      ^
```
I will reconsider.

File: perception/autoware_compare_map_segmentation/src/voxel_grid_map_loader/voxel_grid_map_loader.hpp (hunk 1 above)
Comment (knzo25) [reply perception/autoware_compare_map_segmentation/src/voxel_grid_map_loader/voxel_grid_map_loader.hpp:96]: @yhisaki 
For this one, it is unrelated to your changes (typedef to using), but rather why we are defining
`pcl::Filter<pcl::PointXYZ>::PointCloud` as `PointCloud`. Maybe it is just semantics but I mentioned Dai-san just to be sure

File: sensing/autoware_pointcloud_preprocessor/include/autoware/pointcloud_preprocessor/filter.hpp (hunk 2 above)
Comment (yhisaki) [reply sensing/autoware_pointcloud_preprocessor/include/autoware/pointcloud_preprocessor/filter.hpp]: I realized that this error was due to the implementation of message filter, so I changed pointcloud_preprocessor's cmake to include it as a system library.

**Review by knzo25 — APPROVED** [review]
LGTM for the pointcloud preprocessor part

File: perception/autoware_compare_map_segmentation/src/voxel_grid_map_loader/voxel_grid_map_loader.hpp (hunk 1 above)
Comment (badai-nguyen) [reply perception/autoware_compare_map_segmentation/src/voxel_grid_map_loader/voxel_grid_map_loader.hpp:96]: @knzo25 @yhisaki I'm sorry, I was my typo. I will fix it right after this PR merged.

**Comment by badai-nguyen** [comment]
@yhisaki could you update branch and fix the conflicts?

**Comment by yhisaki** [comment]
@badai-nguyen 
I resolved the conflicts!

**Review by badai-nguyen — APPROVED** [review]
LGTM

File: perception/autoware_compare_map_segmentation/src/voxel_grid_map_loader/voxel_grid_map_loader.hpp (hunk 1 above)
Comment (badai-nguyen) [reply perception/autoware_compare_map_segmentation/src/voxel_grid_map_loader/voxel_grid_map_loader.hpp:96]: > Maybe it is just semantics but I mentioned Dai-san just to be sure

@knzo25 the reason is that I was following defination at https://pointclouds.org/documentation/filters_2include_2pcl_2filters_2voxel__grid_8h_source.html (line 217 ).
Anyway, I agree that it's better to rename it and make a change at https://github.com/autowarefoundation/autoware.universe/pull/9181
Could I as your review on that?

**Pull request merged**